### DISCUSSION TITLE
Domain-specific routes

### DISCUSSION THREAD

**Discussion by marcospassos** [vercel/next.js#62328] [Help]
### Summary

Is there a way to limit some pages to a specific domain only? We have two domains, example.com, and mkt.example.com, and we want to restrict access to the pages under `/page/*` from example.com. We want to allow access to these pages only from the mkt.example.com domain without any redirect. Is that possible?

**Comment by gustavostz** [comment]
For domain-specific routing in Next.js, you can use middleware. Example:

pages/page/_middleware.js:

```
import { NextResponse } from 'next/server';

export function middleware(req) {
  const { hostname } = req.nextUrl;
  if (hostname.startsWith('mkt.')) {
    return NextResponse.next();
  }
  return new Response('Access Denied', { status: 403 });
}
```

This code checks if the request is coming from mkt.example.com. If it is, it allows access. Otherwise, it denies access with a 403 status code.
